jans-lock/cedarling/cedarling/src/authz/mod.rs

```diff
@@ -8,3 +8,55 @@
 //! Part of Cedarling that main purpose is:
 //! - evaluate if authorization is granted for *user*
 //! - evaluate if authorization is granted for *client*
+
+use crate::log::LogStorage;
+use crate::log::{init_logger, log_strategy::LogStrategy, LogWriter};
+use crate::models::log_entry::{LogEntry, LogType};
+use crate::BootstrapConfig;
+use uuid7::{uuid4, Uuid};
+
+/// Authorization Service
+/// The primary service of the Cedarling application responsible for evaluating authorization requests.
+/// It leverages other services as needed to complete its evaluations.
+#[allow(dead_code)]
+pub struct Authz {
+    log_service: LogStrategy,
+    pdp_id: Uuid,
+    application_name: String,
+}
+
+impl Authz {
+    /// Create a new Authorization Service
+    pub fn new(config: BootstrapConfig) -> Self {
+        let log = init_logger(config.log_config);
+        // we use uuid v4 because it is generated based on random numbers.
+        let pdp_id = uuid4();
+        let application_name = config.application_name;
+
+        log.log(
+            LogEntry::new_with_data(pdp_id, application_name.clone(), LogType::System)
+                .set_message("Cedarling Authz initialized successfully".to_string()),
+        );
+
+        Self {
+            log_service: log,
+            pdp_id,
+            application_name,
+        }
+    }
+
+    /// return logs and remove them from the storage
+    pub fn pop_logs(&self) -> Vec<LogEntry> {
+        self.log_service.pop_logs()
+    }
+
+    /// get specific log entry
+    pub fn get_log_by_id(&self, id: &str) -> Option<LogEntry> {
+        self.log_service.get_log_by_id(id)
+    }
+
+    /// returns a list of all log ids
+    pub fn get_log_ids(&self) -> Vec<String> {
+        self.log_service.get_log_ids()
```

Comment:
I don't understand why this is in the Authz engine.

Reply:
Created `Cedarling` instance
and removed from `Authz`